pkgs/top-level/python-packages.nix

```diff
@@ -8892,6 +8892,8 @@ self: super: with self; {
 
   oras = callPackage ../development/python-modules/oras { };
 
+  orbax-checkpoint = callPackage ../development/python-modules/orbax-checkpoint { };
```

Comment:
@GaetanLepage perhaps out of scope for this PR, but orbax-checkpoint's dependency on protobuf causes a conflict with tensorflow since tensorflow has its own protobuf version. This means python packages can't depend on orbax-checkpoint and tensorflow at the same time. See `tensorflow-build` in [pkgs/top-level/python-packages.nix](https://github.com/NixOS/nixpkgs/blob/d8fe5e6c92d0d190646fb9f1056741a229980089/pkgs/top-level/python-packages.nix#L14608-L14612).

```nix
  tensorflow-build = let
    compat = rec {
      protobufTF = pkgs.protobuf_21.override {
        abseil-cpp = pkgs.abseil-cpp_202301;
      };
  ...
```

I first noticed this when trying to package [gpjax](https://github.com/JaxGaussianProcesses/GPJax) with the following [code](https://github.com/stephen-huan/sensor-placement/blob/master/python/pkgs/default.nix).

```shell
nix why-depends --all .#gpjax /nix/store/cb1gs888vfqxawvc65q1dk6jzbayh3wz-source#protobuf_21
```

```text
/nix/store/icb03rmfzgph4bhxm4wic934qwx7y8j7-python3.11-gpjax-0.8.2
├───/nix/store/n8blzy5lqs9gl7yi4m8q5c44gnfkdgcc-python3.11-orbax-checkpoint-0.5.7
│   └───/nix/store/vq7w8s4acdn9kx3638pz186ar0wsh5b9-python3.11-protobuf-4.21.12
│       └───/nix/store/swx2q314if8j37ryn35nr2yb60axmrxz-protobuf-21.12
└───/nix/store/3smllqadx197wmspwd3zkyrxz2rrj01b-python3.11-tensorflow-probability-0.21.0
    └───/nix/store/b6fvycgjpin2g72mfv5fw3f9agqhn697-python3.11-tensorflow-2.13.0
        ├───/nix/store/vq7w8s4acdn9kx3638pz186ar0wsh5b9-python3.11-protobuf-4.21.12
        ├───/nix/store/api4113pl9qg9296cf15igrfnxyifjhl-python3.11-tensorboard-2.16.2
        │   ├───/nix/store/vq7w8s4acdn9kx3638pz186ar0wsh5b9-python3.11-protobuf-4.21.12
        │   └───/nix/store/d4dyq3y1scsaprv1y72fvz7sbn31dqjb-python3.11-tensorboard_plugin_profile-2.11.1
        │       └───/nix/store/vq7w8s4acdn9kx3638pz186ar0wsh5b9-python3.11-protobuf-4.21.12
        └───/nix/store/d4dyq3y1scsaprv1y72fvz7sbn31dqjb-python3.11-tensorboard_plugin_profile-2.11.1
```

It doesn't seem `tensorflow-build` exposes `protobuf-pythonTF` ergonomically but 
```nix
orbax-checkpoint = callPackage ../development/python-modules/orbax-checkpoint { };
```
could still be replaced with
```nix
orbax-checkpoint = callPackage ../development/python-modules/orbax-checkpoint {
  protobuf = pkgs.protobuf_21.override {
    abseil-cpp = pkgs.abseil-cpp_202301;
  };
}
```
with the caveat that the two have to stay in sync. I'm a bit confused what the current situation is, for example, `pkgs.python3Packages.tensorboard` also conflicts with tensorflow which seems strange to me. Why does tensorflow carry its own versions of packages that are also exposed at the top level?